plugins/woocommerce/src/Utilities/PluginUtil.php

```diff
@@ -195,20 +199,29 @@ public function generate_incompatible_plugin_feature_warning( string $feature_id
 		$incompatibles      = array_filter( $incompatibles, 'is_plugin_active' );
 		$incompatibles      = array_values( array_diff( $incompatibles, $this->get_plugins_excluded_from_compatibility_ui() ) );
 		$incompatible_count = count( $incompatibles );
+
+		$feature_warnings = array();
+		if ( 'custom_order_tables' === $feature_id && 'yes' === get_option( 'woocommerce_api_enabled' ) ) {
+			if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
```

Comment:
same as for dismissable notice, I think we only check for the `woocommerce_api_enabled` option, and make this warning filterable. then we can filter it, or remove it, or add a link to the doc page etc without having to push code in this WooCommerce.